deployment/scripts/uuid_migrate.sh

```
@@ -0,0 +1,114 @@
+
+    echo Migration for project $1: $output
+}
+
+function create_context() {
+
+# there's no good way for oc to filter the list of secrets; and there can be several token secrets per SA.
+  # following template prints all tokens for aggregated-logging-fluentd; --sort-by will order them earliest to latest, we will use the last.
+  local sa_token_secret_template='{{range .items}}{{if eq .type "kubernetes.io/service-account-token"}}{{if eq "aggregated-logging-fluentd" (index .metadata.annotations "kubernetes.io/service-account.name")}}{{.data.token}}
+{{end}}{{end}}{{end}}'
+  local failure="false"
+  local nodes_active="false"
+  local output=""
+
+  # check that the aggregated-logging-fluentd SA exists and we can get its token
+  if ! output=$(oc get secret --namespace="${project}" --sort-by=metadata.resourceVersion --template="$sa_token_secret_template" 2>&1); then
+    echo "Error getting aggregated-logging-fluentd service account token; is the master running and are credentials working? Error from oc get secrets follows:"
+    echo -n "$output"
+    return 1
+  elif [[ -z "$output" ]]; then
+    echo "Could not find aggregated-logging-fluentd service account token in $namespace; does it exist?"
+    return 1
+  fi
+  local token=$(echo -e "$output" | tail -1 | base64 -d)
+  # set up a config context using the aggregated-logging-fluentd account and most recent token
+  oc config set-credentials aggregated-logging-fluentd-account \
+    --token="$token" >& /dev/null
+  oc config set-context aggregated-logging-fluentd-context \
+    --cluster=master \
+    --user=aggregated-logging-fluentd-account \
+    --namespace="${project}" >& /dev/null
+
+  oc config use-context aggregated-logging-fluentd-context
+}
+
+function recreate_admin_certs(){
+
+# note: following mess is because we want the error output from the first failure, not a pipeline
```

that doesn't preserve the output from the first failure or prevent further piping; it just keeps feeding it down the line. `-o pipefail` just changes the RC to ensure failures aren't completely swallowed.